collection-scripts/gather_network_logs

```diff
@@ -26,6 +26,60 @@ if [ "$NETWORK_TYPE" == "openshiftsdn" ]; then
         oc -n openshift-sdn exec $OVS_POD -- bash -c "cat /var/log/openvswitch/ovsdb-server.log" > ${NETWORK_LOG_PATH}/${OVS_POD}_ovsdb_log &
         PIDS+=($!)
     done
+elif [ "$NETWORK_TYPE" == "kuryr" ]; then
+      CONTROLLER_POD=$(oc -n openshift-kuryr get pods --no-headers -o custom-columns=':metadata.name' -l app=kuryr-controller | tail -1)
+      oc -n openshift-kuryr logs $CONTROLLER_POD > \
+      ${NETWORK_LOG_PATH}/${CONTROLLER_POD}_log &
+      PIDS+=($!)
+
+      CONTROLLER_RESTART_COUNT="${@:-$(oc get po $CONTROLLER_POD -n openshift-kuryr -o jsonpath={.status.containerStatuses[*].restartCount})}"
+      if [ "$CONTROLLER_RESTART_COUNT" -ne "0" ]; then
+          oc -n openshift-kuryr logs $CONTROLLER_POD --previous > ${NETWORK_LOG_PATH}/${CONTROLLER_POD}_previous_log &
+          PIDS+=($!)
+      fi
+
+      NODES="${@:-$(oc get nodes --no-headers -o custom-columns=':metadata.name')}"
+      for NODE in $NODES; do
+          CNI_POD=$(oc -n openshift-kuryr get pods --no-headers -o custom-columns=":metadata.name" --field-selector spec.nodeName=$NODE -l app=kuryr-cni)
+          oc -n openshift-kuryr logs $CNI_POD > ${NETWORK_LOG_PATH}/${CNI_POD}_log &
+          PIDS+=($!)
+          RESTART_COUNT="${@:-$(oc get po $CNI_POD -n openshift-kuryr -o jsonpath={.status.containerStatuses[*].restartCount})}"
+          if [ "$RESTART_COUNT" -ne "0" ]; then
+              oc -n openshift-kuryr logs $CNI_POD --previous > ${NETWORK_LOG_PATH}/${CNI_POD}_previous_log &
+              PIDS+=($!)
+          fi
+          oc exec -n openshift-kuryr $CNI_POD -- bash -c 'for pid in $(find /host_proc/[1-9]*/ns/net | cut -d/ -f3); do\
+              echo "Namespace: $pid";\
+              nsenter --net=/host_proc/$pid/ns/net ip -d addr;\
+              done' > ${NETWORK_LOG_PATH}/${CNI_POD}_interfaces & PIDS+=($!)
+      done
+
+      oc get pods -A -o wide > ${NETWORK_LOG_PATH}/get_pods & PIDS+=($!)
```

Comment:
We will likely already collect this output with other must-gather runs; I am not sure why we would re-collect the pods list here.

Reply:
When Kuryr-Kubernetes is used we only collect it here. It could be useful to verify if the pods are  stuck on `containerCreating`

Reply:
Let's let the other 'collectors' collect this, and if you want this you simply need to supply `gather; gather_network_logs` as the 'command' to must-gather call; thus getting you all this information in one archive.

Reply:
Seems the `gather` script does list pods running on namespaces created by the user, only the OpenShift ones created by the installer. As it lists the current and previous logs of the Kuryr pods, I'm removing the logs we're fetching from those pods.